src/Repository/InformationProductRepository.php

```diff
@@ -54,6 +55,22 @@ public function findByDigitalResourceTypeDescriptor(DigitalResourceTypeDescripto
         return $qb->getQuery()->getResult();
     }
 
+    /**
+     * Get Information Products by Product Type Descriptor.
+     *
+     * @param InformationProductTypeDescriptor $productTypeDescriptor
+     *
+     * @return array An array of Information Products.
+     */
+    public function findByProductTypeDescriptor(InformationProductTypeDescriptor $productTypeDescriptor): array
+    {
+        $qb = $this->createQueryBuilder('ip');
+        $qb->setParameter('productTypeDescriptor', $productTypeDescriptor);
+        $qb->where($qb->expr()->isMemberOf(':productTypeDescriptor', 'ip.informationProductTypeDescriptors'));
+
+        return $qb->getQuery()->getResult();
```

Comment:
Does this always return array even if it doesn't find anything?

Reply:
It should yes. An empty one.